spec/api/payment/find_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Afterpay::API::Payment::Find do
+  describe '#call' do
+    let(:order_id) { '12345' }
+
+    context 'with proper params' do
+      let(:raw_response) { JSON.generate(paymentState: 'VOIDED') }
+
+      before(:each) do
+        stub_request(:get, %r{api.us.afterpay.com/v2/payments/#{order_id}})
+          .to_return(
+            status: 201,
+            body: raw_response,
+            headers: { 'Content-Type' => 'application/json' }
+          )
+      end
+
+      it 'returns hashie mash' do
+        expect(described_class.call(order_id: order_id).body).to be_instance_of Hashie::Mash
+      end
+
+      it 'returns afterpay order id' do
+        expect(described_class.call(order_id: order_id).body).to include(:paymentState)
+      end
+    end
+
+    context 'without credentials' do
+      let(:error_message) { 'Credentials are required to access this resource.' }
```

Comment:
Instead of this error, I think it's better to test for the `No Afterpay Order was found matching the orderId provided.` error

Reply:
Done!